src/VisualStudio/Core/Def/Implementation/ProjectSystem/MetadataReferences/VisualStudioMetadataReference.cs

```diff
@@ -74,7 +72,7 @@ public void Dispose()
 
         public void UpdateSnapshot()
         {
-            _currentSnapshot = new Snapshot(_provider, Properties, this.FilePath);
+            _currentSnapshot = new Snapshot(_provider, Properties, this.FilePath, _fileChangeTracker);
```

Comment:
❓ Why is this one provided, but the one in `VisualStudioMetadataReferenceManager` is not?

Reply:
We only need to create a file change tracker if we are watching the file for updates, and we only need ensure we are watching files once we've actually observed the contents. VisualStudioMetadataReference provides that service. Directly creating a snapshot doesn't need no file watching by it's definition.